plugins/Jobs.cpp

```diff
@@ -1,1240 +1,1386 @@
+    verifyAttributeInt64(command.request, "jobID", 1);
+    int64_t jobID = command.request.calc64("jobID");
+    string tableName = getTableName(jobID);
 
-        return true; // Successfully processed
+    // Verify there is a job like this and it's not running
+    SQResult result;
+    if (!db.read("SELECT state "
+                 "FROM " + tableName + " "
+                 "WHERE jobID=" + SQ(jobID) + ";",
+                 result)) {
+        STHROW("502 Select failed");
+    }
+    if (result.empty()) {
+        STHROW("404 No job with this jobID");
+    }
+    if (result[0][0] == "RUNNING") {
+        STHROW("405 Can't delete a RUNNING job");
+    }
+    if (result[0][0] == "PAUSED") {
+        STHROW("405 Can't delete a parent job with children running");
     }
 
-    // ----------------------------------------------------------------------
-    else if (SIEquals(requestVerb, "GetJob") || SIEquals(requestVerb, "GetJobs")) {
-        // If we're here it's because peekCommand found some data; re-execute
-        // the query for real now.  However, this time we will order by
-        // priority.  We do this as three separate queries so we only have one
-        // unbounded column in each query.  Additionally, we wrap each inner
-        // query in a "SELECT *" such that we can have an "ORDER BY" and
-        // "LIMIT" *before* we UNION ALL them together.  Looks gnarly, but it
-        // works!
-        SQResult result;
-        const list<string> nameList = SParseList(request["name"]);
-        string safeNumResults = SQ(max(request.calc("numResults"),1));
-        bool mockRequest = command.request.isSet("mockRequest") || command.request.isSet("getMockedJobs");
-        string selectQuery;
-        if (request.isSet("jobPriority")) {
-            selectQuery =
-                "SELECT jobID, name, data, parentJobID, retryAfter, created, repeat, lastRun, nextRun "
-                "FROM jobs "
-                "WHERE state IN ('QUEUED', 'RUNQUEUED') "
-                    "AND priority=" + SQ(request.calc("jobPriority")) + " "
-                    "AND " + SCURRENT_TIMESTAMP() + ">=nextRun "
-                    "AND +name " + (nameList.size() > 1 ? "IN (" + SQList(nameList) + ")" : "GLOB " + SQ(request["name"])) + " " +
-                    string(!mockRequest ? " AND JSON_EXTRACT(data, '$.mockRequest') IS NULL " : "") +
-                "ORDER BY nextRun ASC LIMIT " + safeNumResults + ";";
-        } else {
-            selectQuery =
-                "SELECT jobID, name, data, parentJobID, retryAfter, created, repeat, lastRun, nextRun FROM ( "
-                    "SELECT * FROM ("
-                        "SELECT jobID, name, data, priority, parentJobID, retryAfter, created, repeat, lastRun, nextRun "
-                        "FROM jobs "
-                        "WHERE state IN ('QUEUED', 'RUNQUEUED') "
-                            "AND priority=1000 "
-                            "AND " + SCURRENT_TI
```

Comment:
Here too? You got the tableName by jobID but you need to get it by the parentJobID